packages/dx-react-grid/docs/guides/detail-row-editing.md

```diff
@@ -0,0 +1,232 @@
+# React Grid - Edit Data in the Detail Row
+
+The [detail row](LINK_NEEDED) is an expandable pane that displays data row details. The detail row does not allow users to edit data out of the box, but you can create a custom plugin to implement this functionality. Before you start, peruse the following help topics:
+
+- [Core Concepts](../../../core/docs/guides/fundamentals.md)
+- [Custom Plugin Development](./custom-plugin-development.md)
+
+The implementation consists of the following steps:
+
+1. [Add a Custom Plugin](LINK_NEEDED)
+1. [Handle the Detail Row Expand and Collapse Events](LINK_NEEDED)
+1. [Handle Data Edits](LINK_NEEDED)
+1. [Add the Edit Form](LINK_NEEDED)
+1. [Replace the Expand Icon](LINK_NEEDED)
+
+## Add a Custom Plugin
+
+For information on custom plugins, refer to the following help topic: [Custom Plugin Development](./custom-plugin-development.md). In the following code, the plugin is called `DetailEditCell`, but you can rename it if required.
+
+```jsx
+const DetailEditCell = () => (
+  <Plugin name="DetailEditCell">
+  </Plugin>
+)
+```
+
+## Handle the Detail Row Expand and Collapse Events
+
+When a user expands or collapses a detail row, the [`TableDetailRow`](LINK_NEEDED) plugin executes the `toggleDetailRowExpanded` action. Override this action to switch an expanded row to the edit state or switch a collapsed row back to the normal state.
+
+The IDs of expanded rows are included if the `expandedDetailRowIds` array. To find whether a row is expanded or collapsed, check if this array includes the row ID.
```

Comment:
The IDs of expanded rows are included **IF** the `expandedDetailRowIds` array.

if ???